dashboard/db/migrate/20200316201644_create_contact_rollups_pardot_memory.rb

```diff
@@ -0,0 +1,17 @@
+class CreateContactRollupsPardotMemory < ActiveRecord::Migration[5.0]
+  def change
+    create_table :contact_rollups_pardot_memory do |t|
+      t.string :email, null: false
+      t.integer :pardot_id
+      t.datetime :pardot_id_updated_at
+      t.json :data_synced
+      t.datetime :data_synced_at
+      t.datetime :data_rejected_at
+      t.string :data_rejected_reason
+      t.timestamps
+    end
+
+    add_index :contact_rollups_pardot_memory, :email, unique: true
+    add_index :contact_rollups_pardot_memory, :pardot_id, unique: true
```

Comment:
How does this work for null-able field?

Reply:
[MySQL docs](https://dev.mysql.com/doc/refman/8.0/en/create-index.html) say it allows multiple null values:

> A UNIQUE index creates a constraint such that all values in the index must be distinct. An error occurs if you try to add a new row with a key value that matches an existing row. If you specify a prefix value for a column in a UNIQUE index, the column values must be unique within the prefix length. A UNIQUE index permits multiple NULL values for columns that can contain NULL.

Reply:
I see. This index could be helpful to quickly validate new data and find the highest Pardot ID.

Reply:
FYI, the mysql link above is version 8.0 while the version we have in Aurora is 5.7.12. The part we care about is mostly the same between 2 versions.

Reply:
Ah, good catch thanks!